wandb/sdk/internal/internal_api.py

```diff
@@ -66,18 +66,37 @@ class CreateArtifactFileSpecInput(TypedDict, total=False):
         md5: str
         mimetype: Optional[str]
         artifactManifestID: Optional[str]  # noqa: N815
+        uploadPartsInput: Optional["UploadPartsInput"]
 
     class CreateArtifactFilesResponseFile(TypedDict):
         id: str
         name: str
         displayName: str  # noqa: N815
         uploadUrl: Optional[str]  # noqa: N815
         uploadHeaders: Sequence[str]  # noqa: N815
+        multipartUploadUrls: Optional[dict[str, any]]
         artifact: "CreateArtifactFilesResponseFileNode"
 
     class CreateArtifactFilesResponseFileNode(TypedDict):
         id: str
 
+    class UploadPartsInput(TypedDict):
```

Comment:
Does this run on Python<=3.7? I thought TypedDict was a relatively recent introduction. If you do need typing it's absolutely fine to use a `dataclass` here. It's probably also fine to leave it untyped.

Reply:
I was getting a bunch of mypy errors without it so i added the dataclasses like the existing code has